Fixes #20638: Document bulk create support in OpenAPI schema

jnovinger opened the pull request:
### Fixes: #20638

POST operations on `NetBoxModelViewSet` endpoints accept both single objects and arrays, but the schema only documented single objects. This prevented REST API client generators from producing correct code.

Add explicit bulk_create_enabled flag to `NetBoxModelViewSet` and update schema generation to emit `oneOf` for these endpoints.

FWIW, I don't believe the `PUT` issue reported is a bug. The schema seems to correctly document `PUT` as array-only. NetBox's PUT is a bulk operation by design. For single updates, use `PATCH /api/model/{id}/`.

jnovinger committed:
Fixes #20638: Document bulk create support in OpenAPI schema

POST operations on NetBoxModelViewSet endpoints accept both single
objects and arrays, but the schema only documented single objects.
This prevented API client generators from producing correct code.

Add explicit bulk_create_enabled flag to NetBoxModelViewSet and
update schema generation to emit oneOf for these endpoints.

jeremystretch commented on netbox/core/api/schema.py:
```diff
@@ -128,6 +159,35 @@ def get_response_serializers(self) -> typing.Any:
 
         return response_serializers
 
+    def _get_request_for_media_type(self, serializer, direction):
```
To preserve the default
```suggestion
    def _get_request_for_media_type(self, serializer, direction='request'):
```

jnovinger committed:
Address PR feedback

- Removed brittle serializer marking mechanism in favor of direct checks
  on behavior.
- Attempted to introduce a bulk_create action and then route to it on
  POST in NetBoxRouter, but ran in to several obstacles including
  breaking HTTP status code reporting in the schema. Opted to simply

jeremystretch reviewed (changes requested):
Assuming we can remove the now-unused `bulk_create_enabled` var I think this is good to go.

jnovinger committed:
Remove unused bulk_create_enabled attr

Co-authored-by: Jeremy Stretch <jstretch@netboxlabs.com>
